src/main/java/org/agbrothers/englishbot/processing/ProcessingCore.java

```diff
@@ -23,46 +24,50 @@
 public class ProcessingCore {
     private TelegramBot telegramBot;
     private LessonService lessonService;
-    private final Map<String, String> stateByChatId = new HashMap<>();
-    private ProcessorHolder processorHolder;
+    private ProcessorFactory processorFactory;
+    private UserService userService;
 
     public void processUserRequest(String chatId, String messageText) {
-        String stateId = getStateId(chatId);
-        if (stateId == null) {
-            sendHelloMessage(chatId);
-            stateByChatId.put(chatId, State.MAIN_MENU);
+        User user = userService.containsUser(chatId)
+                ? userService.getUserByChatId(chatId)
+                : new User(chatId);
+
+        if (user.getStateId() == null) {
+            sendHelloMessage(user.getChatId());
+            user.setStateId(State.MAIN_MENU);
         }
 
         switch (messageText) {
```

Comment:
move this switch statement to the separate method named applyUserSate or something like that